zio-http/shared/src/main/scala/zio/http/internal/FormState.scala

```diff
@@ -29,8 +29,10 @@ private[http] object FormState {
 
   final class FormStateBuffer(boundary: Boundary) extends FormState { self =>
 
-    private val tree0: ChunkBuilder[FormAST] = ChunkBuilder.make[FormAST]()
-    private val buffer: ChunkBuilder.Byte    = new ChunkBuilder.Byte
+    private val tree0: ChunkBuilder[FormAST]    = ChunkBuilder.make[FormAST]()
+    private val buffer: ChunkBuilder[Byte]      = new ChunkBuilder.Byte
+    private var bufferSize: Int                 = 0
```

Comment:
I was going to link [this PR](https://github.com/zio/zio/pull/8892) but just realised that it's only for Scala 2.13+.. And also that it was you that opened it in the first place 😅

Reply:
Yes, unfortunately I was not able to add Scala 2.12 support for the knownSize of a ChunkBuilder.